compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -4274,19 +4278,26 @@ private boolean equalityIntersectionExists(Set<BType> lhsTypes, Set<BType> rhsTy
         return matchFound;
     }
 
-    boolean validNumericTypeExists(BType type) {
-
+    boolean validNumericStringOrXmlTypeExists(BType type, boolean isNumericType) {
```

Comment:
Is this the cleanest way to do this? Looks like we are mixing up multiple things. Why can't we have two separate methods for numeric vs string/xml?

Reply:
The `isNumericType` param name is also misleading IMO. Reads like it indicates if `type` is a numeric type, but that's now what it represents.

Reply:
Create separate methods